src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -666,13 +670,21 @@
   (-plan->sql [query]
     (let [has-where? (boolean (:where query))
           has-projections? (not (empty? (:projected-fields query)))
-          update-when (fn [m pred ks f]
-                        (if pred
-                          (update-in m ks f)
-                          m))
+          update-hash-fields
```

Comment:
Alright, hmm. This now bothers me perhaps. I wonder how often we call this during these decisions as well, postgres? is not memoized, I don't think it calls the db per se, it just grabs the JDBC meta data. I guess the only way around this is some sort of threading of db type instead ... hmm.